File: SpringbootAngular/frontend/src/model/search.request.ts
export class SearchRequest {
  productId: number;
  productName: string;
  categoryId: number;
  price: number;

  constructor(productId?: number, productName?: string, categoryId?: number, price?: number) {
    this.categoryId = productId;
    this.productName = productName;
    this.categoryId = categoryId;
    this.price = price;
  }
}

File: SpringbootAngular/backend/src/main/java/com/spring/angular/helper/SearchRequest.java
package com.spring.angular.helper;

public class SearchRequest {

    String productName;
    String categoryId;
    int price;

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public String getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(String categoryId) {
        this.categoryId = categoryId;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }
}
